src/main/drivers/accgyro/accgyro_spi_icm426xx.c

```diff
@@ -168,10 +176,61 @@ static aafConfig_t aafLUT42605[AAF_CONFIG_COUNT] = {  // see table in section 5.
     [AAF_CONFIG_1962HZ] = { 63, 3968,  3 }, // 995 Hz is the max cutoff on the 42605
 };
 
+#if defined(USE_GYRO_CLKIN)
+static pwmOutputPort_t pwmGyroClk = {0};
+
+static bool initExternalClock(void)
+{
+    const ioTag_t tag = IO_TAG(GYRO_CLKIN_PIN);
+    const IO_t io = IOGetByTag(tag);
+
+    pwmGyroClk.io = io;
+    pwmGyroClk.enabled = true;
+
+    const timerHardware_t *timer = timerAllocate(tag, OWNER_GYRO_CLKIN, 0);
+    if (!timer) {
+        // Error handling: failed to allocate timer
+        return false;
+    }
+
+    IOInit(io, OWNER_GYRO_CLKIN, 0);
+    IOConfigGPIOAF(io, IOCFG_AF_PP, timer->alternateFunction);
+
+    uint32_t pwmFrequency = 32000;  // PWM frequency set to 32 kHz
+    const uint32_t clock = timerClock(timer->tim);  // Get the timer clock frequency
```

Comment:
Do any other gyros support this? Should this frequency be fixed or have a setting?

Reply:
@SteveCEvans clock input from 31kHz to 50kHz for ICM42688P and ICM42605